docs/reference/using/python/api/fastapi2.rst

```diff
@@ -0,0 +1,311 @@
+      # remove the member from the team
+      team = person.team
+
+      if team is None:
+          return 'Person is not in a team'
+      else:
+          team.members.remove(person)
+          await db.save(team)
+          return f'{pname} quit {team.name}'
+
+Let's setup some GameSession endpoints. Find all the games where members of a specific team are playing, that aren't full and that are in "Waiting" status:
+
+.. code-block:: python
+
+  class GameSessionBase(default.GameSession.__variants__.Base):
+      title: default.GameSession.__typeof__.title
+      status: default.GameSession.__typeof__.status
+      num_players: default.GameSession.__typeof__.num_players
+      is_full: default.GameSession.__typeof__.is_full
+
+
+  @app.get("/games/", response_model=list[GameSessionBase])
+  async def get_games_with_team(team_name: str):
+      db = g.client
+      q = default.GameSession.filter(
+          # use an expression as a filter
+          lambda g: std.any(g.players.team.name == team_name),
+          # filter by status and is_full
+          status=default.GameStatus.Waiting,
+          is_full=False,
+      ).order_by(title=True)
+
+      return await db.query(q)
+
+Next let's setup an endpoint that starts up a game if the game is ready to be started. To start the game session needs to be full and currently in "Waiting" status. Additionally, we will use a single ``update`` query to do this instead of fetching the GameSession object, then checking the conditions in Python and saving the updated value:
+
+.. code-block:: python
+
+  @app.post("/games/{game_id}/start", response_model=str)
+  async def start_game(game_id: uuid.UUID):
+      db = g.client
+      # instead of fetching the game, then updating and saving,
+      # we can update directly
+      q = default.GameSession.filter(
+          id=game_id,
+          # make sure the game is eligible to be started
+          status=default.GameStatus.Waiting,
+          is_full=True,
+      ).update(
+          status='Active',
+      ).select('*')  # select the updated game
+      result = await db.query(q)
+
+      if len(result) == 0:
+          return "Game not started"
+      else:
+          return f"{result[0].num_players} player game started"
+
+We can also make an endpoint for finding GameSessions that have enough room for an entire Team. Just like in the previous example, we want to do this in a single query instead of fetching the Team data first and then picking the GameSessions based on that:
+
+.. code-block:: python
+
+  @app.get("/games/fit_team", response_model=list[GameSessionBase])
+  async def get_games_for_team(team_name: str):
+      db = g.client
+      # make the team subquery
+      team = default.Team.filter(name=team_name)
+      # use the team subquery to find the games
+      q = default.GameSession.filter(
+          lambda g: g.max_players - g.num_players >= std.count(team.members),
+      ).order_by(title=True)
+
+      return await db.query(q)
```

Comment:
Oh, this is fancy! Putting on my RESTifarian glasses, I'd write this as a `fit_team_name` attribute on the `GameFilter` from earlier. We'd also want to update it to take into account games that already have some members of this team.